lib/validator/validator.go

```diff
@@ -364,6 +364,14 @@ func validateRule(v *validator.Validate, structLevel *validator.StructLevel) {
 				"Destination.NotPorts", "", reason("protocol does not support ports"))
 		}
 	}
+
+	if rule.Source.Net != nil && rule.Destination.Net != nil {
+		if (rule.Source.Net.IP.To4() == nil && rule.Destination.Net.IP.To4() != nil) ||
```

Comment:
I think you can just do  `rule.Source.Net.Version() != rule.Desintation.Net.Version()`